src/audio/module_adapter/module_adapter.c

```diff
@@ -879,23 +877,30 @@ static int module_adapter_audio_stream_type_copy(struct comp_dev *dev)
 		sink = container_of(blist, struct comp_buffer, source_list);
 		sinks_c[i++] = buffer_acquire(sink);
 	}
+	num_output_buffers = i;
+
 	i = 0;
 	list_for_item(blist, &dev->bsource_list) {
 		struct comp_buffer *source;
 
 		source = container_of(blist, struct comp_buffer, sink_list);
 		source_c[i++] = buffer_acquire(source);
 	}
+	num_input_buffers = i;
 
 	/* setup active input/output buffers for processing */
-	if (mod->num_output_buffers == 1) {
+	if (num_output_buffers == 1) {
 		num_input_buffers = module_single_sink_setup(dev, source_c, sinks_c);
 		if (sinks_c[0]->sink->state == dev->state)
 			num_output_buffers = 1;
+		else
+			num_output_buffers = 0;
```

Comment:
if (sinks_c[0]->sink->state ！= dev->state)
        num_output_buffers = 0;

4 lines to 2 lines.